### PR TITLE
[move-compiler] Support underscores in number literals

### PR DESCRIPTION
**Pull request by macalinao**
## Motivation

Easier to read groupings of decimals.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

Yes

## Test Plan

Add tests for parsing numbers

### CODE REVIEW

**Review by tnowacki — COMMENTED** [review]
So far looks great! And easy enough :) 

cc @damirka

**Comment by sblackshear** [comment]
Fantastic, this is a much-requested feature! Thanks for this.

File: language/move-command-line-common/src/parser.rs [hunk 1]
```diff
@@ -445,6 +457,7 @@ mod tests {
             "0u64x",
             "0u6 4",
             "0u",
+            "_10",
```
Comment (sblackshear) [inline language/move-command-line-common/src/parser.rs:460]: A few other negative tests might be useful here; e.g., `10_u8`, `5_bool`.

File: language/move-command-line-common/src/parser.rs [hunk 2]
```diff
@@ -385,6 +394,9 @@ mod tests {
             ("18446744073709551615", V::InferredNum(18446744073709551615)),
             ("18446744073709551615u64", V::U64(18446744073709551615)),
             ("0u128", V::U128(0)),
+            ("1_0u8", V::U8(1_0)),
+            ("1_000u64", V::U64(1_000)),
+            ("1_000_000u128", V::U128(1_000_000)),
```
Comment (sblackshear) [inline language/move-command-line-common/src/parser.rs:399]: Perhaps `1_000`, `1_0_0_0u64` as well?

**Review by sblackshear — COMMENTED** [review]
I think @tnowacki should have the final say here, just suggesting a few tests

File: language/move-command-line-common/src/parser.rs (hunk 2 above)
Comment (macalinao) [reply language/move-command-line-common/src/parser.rs:402]: added

File: language/move-command-line-common/src/parser.rs (hunk 1 above)
Comment (macalinao) [reply language/move-command-line-common/src/parser.rs]: done

**Review by damirka — APPROVED** [review]
LGTM!

**Review by tnowacki — CHANGES_REQUESTED** [review]
Could you add compiler specific tests? 

Would like to see several positive/negative tests of the grammar in
language/move-compiler/tests/move_check/parser/
Need to test
- leading underscore
- multiple underscores in a row
- many trailing underscores
- underscores after the `u<size>` suffix


And tests in 
language/move-compiler/transactional-tests/tests/parser/ 
that test the value is the same with/without underscores

**Comment by oxade** [comment]
@macalinao have time to remedy this?

**Comment by oxade** [comment]
Spoke to @macalinao who's busy at the moment.
I'll be taking over the PR and adding the necessary tests.

**Comment by oxade** [comment]
PR continued [here](https://github.com/move-language/move/pull/542)

**Comment by oxade** [comment]
@tnowacki @sblackshear please close this PR as the other one has merged. I dont have right to close it.

**Pull request closed without merging**